api/src/main/java/org/motechproject/nms/api/web/UserController.java

```diff
@@ -116,14 +115,21 @@ public UserResponse getUserDetails(@PathVariable String serviceName,
 
     private UserResponse getKilkariResponseUser(Long callingNumber) {
         KilkariUserResponse user = new KilkariUserResponse();
-        Subscriber subscriber = kilkariService.getSubscriber(callingNumber);
-        if (subscriber == null) {
-            throw new NotFoundException(String.format(NOT_FOUND, CALLING_NUMBER));
-        }
-        Set<Subscription> subscriptions = subscriber.getSubscriptions();
         Set<String> packs = new HashSet<>();
-        for (Subscription subscription : subscriptions) {
-            packs.add(subscription.getSubscriptionPack().getName());
+        Subscriber subscriber = kilkariService.getSubscriber(callingNumber);
+        if (subscriber != null) {
+            Set<Subscription> subscriptions = subscriber.getSubscriptions();
+            for (Subscription subscription : subscriptions) {
+                packs.add(subscription.getSubscriptionPack().getName());
+            }
+        } else {
+            /*
+            No subscriber found in the database, according to https://github.com/koshalt/mim/issues/9#event-288912397
+            we need to respond with an empty subscription list, the circle provider in the request and our guess as
+            to what the defaultLanguageLocationCode is, which is done after this method returns
+            */
+            //todo: figure out the defaultLanguageLocationCode, should that be coming from the language module?
+            user.setDefaultLanguageLocationCode("??");
         }
         user.setSubscriptionPackList(packs);
```

Comment:
This is how we get defaultLanguage in UserController.  I would do the same here.

```java
        Language language = languageService.getDefaultCircleLanguage(circle);
        if (language != null && user != null) {
            user.setDefaultLanguageLocationCode(language.getCode());
        }
```